components/style/values/specified/mod.rs

```diff
@@ -339,47 +367,61 @@ impl ToComputedValue for Angle {
     }
 
     fn from_computed_value(computed: &Self::ComputedValue) -> Self {
-        Angle {
-            radians: computed.radians(),
-            was_calc: false,
-        }
+        Angle::from_radians(computed.radians())
     }
 }
 
 impl Angle {
+    /// Returns an angle with the given value in degrees.
+    pub fn from_degrees(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Degree, was_calc: false }
+    }
+    /// Returns an angle with the given value in gradians.
+    pub fn from_gradians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Gradian, was_calc: false }
+    }
+    /// Returns an angle with the given value in turns.
+    pub fn from_turns(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Turn, was_calc: false }
+    }
+    /// Returns an angle with the given value in radians.
+    pub fn from_radians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Radian, was_calc: false }
```

Comment:
I _think_ we may still need to prevent infinite values from coming here, because Gecko may rely on it even for specified values. I'll do a Gecko try run to double-check though.